tests/test_types/test_document.py

```diff
@@ -73,3 +74,34 @@ def test_annotate_box_groups_gets_text(self):
 
         doc.annotate(new_span_groups=box_groups)
         assert doc.new_span_groups[0].text.startswith("Gutman G, Rosenzweig D, Golan J")
+
+    def test_annotate_box_groups_allocates_all_overlapping_tokens(self):
```

Comment:
Might be worth adding a docstring to this test to indicate it's a regression test to verify annotating box groups onto a doc produces the same output as a last-known-good fixture